test/test_datapipe.py

```diff
@@ -2703,5 +2706,59 @@ def reset(self):
             list(it)
         self.assertEqual(3, datapipe._number_of_samples_yielded)
 
+
+class TestIterDataPipeGraphFastForward(TestCase):
+
+    def _fast_forward_graph_test_helper(self, datapipe, fast_forward_fn, n_iterations=3, rng=None):
+        if rng is None:
+            rng = torch.Generator()
+        initial_rng_state = rng.get_state()
+        torch.utils.data.graph_settings.apply_shuffle_seed(datapipe, rng)
+        expected_full_res = list(datapipe)
+
```

Comment:
Change rng state here (to something totally different) and do one more 'epoch' to make sure we can restore state between epochs.

Reply:
I just added that. Please have a look to see if it is what you meant